tests/test_metadata.py

```diff
@@ -570,6 +570,112 @@ def test_inferred_input_fails(fake_input):
         Metadata.from_infer(fake_input)
 
 
+merge_meta_test = Metadata.from_dict(
+    {
+        "name": "merge_test",
+        "columns": [{"name": "c1", "type": "int64"}, {"name": "c2", "type": "string"}],
+        "partitions": ["c1"],
+    }
+)
+
+merge_meta_diff_col_type = Metadata.from_dict(
+    {
+        "name": "merge_test",
+        "columns": [
+            {"name": "c3", "type": "string"},
+            {"name": "c1", "type": "int64"},
+            {"name": "c2", "type": "float64"},
+        ],
+        "partitions": ["c1"],
+    }
+)
+
+merge_meta_diff_partitions = Metadata.from_dict(
+    {
+        "name": "merge_test",
+        "columns": [
+            {"name": "c3", "type": "string"},
+            {"name": "c1", "type": "int64"},
+            {"name": "c2", "type": "string"},
+        ],
+        "partitions": ["c1", "c3"],
+    }
+)
+
+
+@pytest.mark.parametrize(
+    "m1,m2",
+    [
+        (merge_meta_test, merge_meta_diff_col_type),
+        (merge_meta_test, merge_meta_diff_partitions),
+    ],
+)
+def test_merge_error_raised(m1, m2):
```

Comment:
Can you add another test that verifies that using `mismatch=error` doesn't cause an error when it isn't supposed to

Reply:
https://github.com/moj-analytical-services/mojap-metadata/blob/d29361c105e7006a7b7deef4398eb503e67ea146/tests/test_metadata.py#L618-L656